src/BEditaClient.php

```diff
@@ -357,6 +357,53 @@ public function remove($id) : ?array
         return $this->delete(sprintf('/trash/%s', $id));
     }
 
+    /**
+     * Upload file (POST)
+     *
+     * @param string $filename The file name
+     * @param string $filepath The file full path
+     * @param array $headers Custom request headers
+     * @return array|null Response in array format
+     */
+    public function upload($filename, $filepath, array $headers = ['Content-Type' => '']) : array
+    {
+        $path = sprintf('/streams/upload/%s', $filename);
+        $handle = fopen($filepath, "r");
+        $body = fread($handle, filesize($filepath));
+        fclose($handle);
+        $headers['Content-Type'] = mime_content_type($filepath);
```

Comment:
if `Content-Type` is set in `$headers` its value should be used.
If no `Content-Type` is provided, `mime_content_type` or other strategy may be used.

Not clear from the signature: I'd use `?array $headers = null` like in other cases